chartfx-dataset/src/test/java/de/gsi/dataset/serializer/helper/CmwLightHelper.java

```diff
@@ -0,0 +1,449 @@
+
+        getFieldHeader(ioSerialiser);
+        pojo.int1 = ioSerialiser.getInt();
+        getFieldHeader(ioSerialiser);
+        pojo.int2 = ioSerialiser.getInt();
+
+        getFieldHeader(ioSerialiser);
+        pojo.long1 = ioSerialiser.getLong();
+        getFieldHeader(ioSerialiser);
+        pojo.long2 = ioSerialiser.getLong();
+
+        getFieldHeader(ioSerialiser);
+        pojo.float1 = ioSerialiser.getFloat();
+        getFieldHeader(ioSerialiser);
+        pojo.float2 = ioSerialiser.getFloat();
+
+        getFieldHeader(ioSerialiser);
+        pojo.double1 = ioSerialiser.getDouble();
+        getFieldHeader(ioSerialiser);
+        pojo.double2 = ioSerialiser.getDouble();
+
+        getFieldHeader(ioSerialiser);
+        pojo.string1 = ioSerialiser.getString();
+        if (!TestDataClass.isCmwCompatibilityMode()) { // disabled since reference CMW impl does not support UTF-8
+            getFieldHeader(ioSerialiser);
+            pojo.string2 = ioSerialiser.getString();
+        }
+
+        // 1-dim arrays
+        getFieldHeader(ioSerialiser);
+        pojo.boolArray = ioSerialiser.getBooleanArray();
+        getFieldHeader(ioSerialiser);
+        pojo.byteArray = ioSerialiser.getByteArray();
+        // getFieldHeader(ioSerialiser);
+        // pojo.charArray = ioSerialiser.getCharArray(ioSerialiser);
+        getFieldHeader(ioSerialiser);
+        pojo.shortArray = ioSerialiser.getShortArray();
+        getFieldHeader(ioSerialiser);
+        pojo.intArray = ioSerialiser.getIntArray();
+        getFieldHeader(ioSerialiser);
+        pojo.longArray = ioSerialiser.getLongArray();
+        getFieldHeader(ioSerialiser);
+        pojo.floatArray = ioSerialiser.getFloatArray();
+        getFieldHeader(ioSerialiser);
+        pojo.doubleArray = ioSerialiser.getDoubleArray();
+        getFieldHeader(ioSerialiser);
+        pojo.stringArray = ioSerialiser.getStringArray();
+
+        // multidim case
+        getFieldHeader(ioSerialiser);
+        pojo.nDimensions = ioSerialiser.getIntArray();
+        getFieldHeader(ioSerialiser);
+        pojo.boolNdimArray = ioSerialiser.getBooleanArray();
+        getFieldHeader(ioSerialiser);
+        pojo.byteNdimArray = ioSerialiser.getByteArray();
+        getFieldHeader(ioSerialiser);
+        pojo.shortNdimArray = ioSerialiser.getShortArray();
+        getFieldHeader(ioSerialiser);
+        pojo.intNdimArray = ioSerialiser.getIntArray();
+        getFieldHeader(ioSerialiser);
+        pojo.longNdimArray = ioSerialiser.getLongArray();
+        getFieldHeader(ioSerialiser);
+        pojo.floatNdimArray = ioSerialiser.getFloatArray();
+        getFieldHeader(ioSerialiser);
+        pojo.doubleNdimArray = ioSerialiser.getDoubleArray();
+
+        final WireDataFieldDescription field = getFieldHeader(ioSerialiser);
+        if (field == null) {
+            // reached the end
+            return;
+        }
+
+        if (field.getDataType().equals(de.gsi.dataset.serializer.DataType.START_MARKER)) {
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'de.gsi.dataset.serializer.DataType.START_MARKER' due to existing import 'de.gsi.dataset.serializer.DataType'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6093416)